x-pack/platform/plugins/shared/triggers_actions_ui/public/application/sections/rule_details/components/rule_definition.tsx

```diff
@@ -173,7 +174,10 @@ export const RuleDefinition: React.FunctionComponent<RuleDefinitionProps> = memo
                 >
                   <EuiFlexItem grow={false}>
                     {hasEditButton ? (
-                      <EuiButtonEmpty onClick={onEditRuleClick} flush="left">
+                      <EuiButtonEmpty
+                        onClick={navigateToEditRuleForm ? navigateToEditRuleForm : onEditRuleClick}
```

Comment:
The `navigateToEditRuleForm` should take a `ruleId` so I think it is better if we do something like:

```
const onEditRuleClick = () => {
      if (navigateToEditRuleForm) {
         navigateToEditRuleForm(rule.id);
         return;
      }
     
      // rest of the code as before
      navigateToApp('management', {
        path: `insightsAndAlerting/triggersActions/${getEditRuleRoute(rule.id)}`,
        state: {
          returnApp: 'management',
          returnPath: `insightsAndAlerting/triggersActions/${getRuleDetailsRoute(rule.id)}`,
        },
      });
    };
```

and the button remains as before: `<EuiButtonEmpty onClick={onEditRuleClick} flush="left">`. Same for below.